src/net.cpp

```diff
@@ -342,8 +342,7 @@ CNode* CConnman::ConnectNode(CAddress addrConnect, const char *pszDest, bool fCo
         CNode* pnode = FindNode((CService)addrConnect);
         if (pnode)
         {
-            pnode->AddRef();
-            return pnode;
+            return NULL;
```

Comment:
I think it makes sense to log "already connected" here, otherwise it looks like the attempt fell into a black hole.